File: csharp/SSSPResult.cs
namespace SSSP;

/// <summary>
/// Result of single-source shortest path: distances and predecessor pointers.
/// Distance is double.PositiveInfinity for unreachable vertices.
/// Predecessor is -1 for source and unreachable vertices.
/// </summary>
public sealed class SSSPResult
{
    /// <summary>Distance from source to each vertex.</summary>
    public double[] Distance { get; }

    /// <summary>Predecessor on shortest path; -1 if none (source or unreachable).</summary>
    public int[] Predecessor { get; }

    public int VertexCount => Distance.Length;

    public SSSPResult(int n)
    {
        Distance = new double[n];
        Predecessor = new int[n];
    }
}

File: java/SsspResult.java
package sssp;

/** Result of single-source shortest path. Predecessor is null for source or unreachable. */
public final class SsspResult {
    private final double[] distance;
    private final Integer[] predecessor;

    public SsspResult(double[] distance, Integer[] predecessor) {
        this.distance = distance;
        this.predecessor = predecessor;
    }

    public double[] getDistance() {
        return distance;
    }

    public Integer[] getPredecessor() {
        return predecessor;
    }

    public int vertexCount() {
        return distance.length;
    }
}
